release_notes_generator/record/default_record_factory.py

```
@@ -0,0 +1,165 @@
+
+    def generate(self, github: Github, data: MinedData) -> dict[int | str, Record]:
+        """
+        Generate records for release notes.
+        Parameters:
+            github (GitHub): The GitHub instance to generate records for.
+            data (MinedData): The MinedData instance containing repository, issues, pull requests, and commits.
+        Returns:
+            dict[int|str, Record]: A dictionary of records where the key is the issue or pull request number.
+        """
+
+        def register_pull_request(pull: PullRequest, skip_rec: bool) -> None:
```

Ruff (0.12.2)

Unused function argument: `skip_rec`

(ARG001)